internal/goldendataset/generator_commons.go

```diff
@@ -22,10 +22,73 @@ import (
 
 	"github.com/spf13/cast"
 
-	"go.opentelemetry.io/collector/internal/data"
+	"go.opentelemetry.io/collector/consumer/pdata"
 	otlpcommon "go.opentelemetry.io/collector/internal/data/protogen/common/v1"
 )
 
+func convertMapToAttributeMap(attrsMap map[string]interface{}) pdata.AttributeMap {
+	attributeMap := pdata.NewAttributeMap()
+	if attrsMap == nil {
+		return attributeMap
+	}
+	for key, value := range attrsMap {
+		attributeMap.Insert(key, convertToAttributeValue(value))
+	}
+	return attributeMap
+}
+
+func convertToAttributeValue(value interface{}) pdata.AttributeValue {
+	var newValue pdata.AttributeValue
+	switch val := value.(type) {
+	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64:
+		newValue = pdata.NewAttributeValueInt(cast.ToInt64(val))
+	case float32, float64:
+		newValue = pdata.NewAttributeValueDouble(cast.ToFloat64(val))
+	case bool:
+		newValue = pdata.NewAttributeValueBool(cast.ToBool(val))
+	case *otlpcommon.ArrayValue:
+		newValue = pdata.NewAttributeValueArray()
+		for _, v := range val.GetValues() {
+			newValue.ArrayVal().Append(convertAnyValue(v))
+		}
+	case *otlpcommon.KeyValueList:
+		newValue = pdata.NewAttributeValueMap()
+		for _, kv := range val.GetValues() {
+			newValue.MapVal().Insert(kv.Key, convertAnyValue(kv.Value))
+		}
+	default:
+		newValue = pdata.NewAttributeValueString(val.(string))
+	}
+	return newValue
+}
+
+func convertAnyValue(original otlpcommon.AnyValue) pdata.AttributeValue {
```

Comment:
This still depends on the proto

Reply:
Ok. I fully removed the dependency on proto AnyValue in the construction of the attribute map.